packages/backend-common/src/service/lib/ServiceBuilderImpl.ts

```diff
@@ -14,44 +14,74 @@
  * limitations under the License.
  */
 
+import { ConfigReader } from '@backstage/config';
 import compression from 'compression';
 import cors from 'cors';
 import express, { Router } from 'express';
 import helmet from 'helmet';
 import { Server } from 'http';
 import stoppable from 'stoppable';
 import { Logger } from 'winston';
-import { getRootLogger } from '../logging';
+import { useHotCleanup } from '../../hot';
+import { getRootLogger } from '../../logging';
 import {
   errorHandler,
   notFoundHandler,
   requestLoggingHandler,
-} from '../middleware';
-import { ServiceBuilder } from './types';
-import { useHotCleanup } from '../hot';
+} from '../../middleware';
+import { ServiceBuilder } from '../types';
+import { readBaseOptions, readCorsOptions } from './config';
 
 const DEFAULT_PORT = 7000;
+const DEFAULT_HOST = 'localhost';
```

Comment:
so this file retains defaults AND env reading - eventually we should probably remove the code for both of those and just force the use of config and/or setting env vars through the config system (such as `APP_CONFIG_backend_bindPort=7777`)

Reply:
In a way we're already providing configs through the app-config as well, so I think it could make sense to just require those to be set? Also thinking the one exception we might want to make is PORT, as it's used in some hosting envs. Upon request though :grin:

Reply:
Then again, saying `listen: [localhost:$PORT]` would be better and more explicit then...

Reply:
yeah agree that `listen: [localhost:$PORT]` or something of the form would be way better. Didn't wanna creature feep any interpolation into the config yet, but as a solution for someone that wants to deploy in an env that requires `PORT` to be read, I think we can wait and it's better.